core/data/src/test/java/com/google/samples/apps/nowinandroid/core/data/repository/OfflineFirstNewsRepositoryTest.kt

```diff
@@ -283,4 +287,70 @@ class OfflineFirstNewsRepositoryTest {
                     .sortedBy(NewsResourceTopicCrossRef::toString),
             )
         }
+
+    @Test
+    fun offlineFirstNewsRepository_sends_notifications_for_newly_synced_news_that_is_followed() =
+        testScope.runTest {
+            // User has onboarded
+            niaPreferencesDataSource.setShouldHideOnboarding(true)
+
+            val networkNewsResources = network.getNewsResources()
+
+            val followedTopicIds = networkNewsResources
+                .flatMap(NetworkNewsResource::topicEntityShells)
+                .mapNotNull { topic ->
+                    when (val intTopicId = topic.id.toIntOrNull()) {
+                        null -> null
+                        else -> when (intTopicId % 2) {
+                            0 -> topic.id
+                            else -> null
+                        }
+                    }
+                }
+                .toSet()
+
+            // Set followed topics
+            niaPreferencesDataSource.setFollowedTopicIds(followedTopicIds)
+
+            subject.syncWith(synchronizer)
+
+            // Notifier should have been called with only news resources that have topics
+            // that the user follows
+            assertEquals(
+                expected = networkNewsResources
+                    .filter { (it.topics intersect followedTopicIds).isNotEmpty() }
+                    .map(NetworkNewsResource::id)
+                    .sorted(),
+                actual = notifier.addedNewsResources.first().map(NewsResource::id).sorted(),
+            )
+        }
+
+    @Test
+    fun offlineFirstNewsRepository_does_not_sends_notifications_for_existing_news_resources() =
```

Comment:
Have you considered using backticks syntax for naming tests? 

`fun offlineFirstNewsRepository does not sends notifications for existing news resources`?

Oh, btw, there's a grammar error does_not_send**~s~**

Reply:
Fixed error. I considered it, but the rest of the app uses underscores. I also found out recently that backticks method names are not supported for Kotlin JS.